src/test/java/com/squareup/auto/value/redacted/AutoValueRedactedExtensionTest.java

```diff
@@ -0,0 +1,64 @@
+package com.squareup.auto.value.redacted;
+
+import com.google.auto.value.processor.AutoValueProcessor;
+import com.google.testing.compile.JavaFileObjects;
+import java.util.Arrays;
+import javax.tools.JavaFileObject;
+import org.junit.Before;
+import org.junit.Test;
+
+import static com.google.common.truth.Truth.assertAbout;
+import static com.google.testing.compile.JavaSourcesSubjectFactory.javaSources;
+
+public class AutoValueRedactedExtensionTest {
+  private JavaFileObject redacted;
+
+  @Before public void setUp() {
+    redacted = JavaFileObjects.forSourceString("test.Redacted", ""
+        + "package test;"
+        + "import java.lang.annotation.Retention;"
+        + "import java.lang.annotation.Target;"
+        + "import static java.lang.annotation.ElementType.FIELD;"
+        + "import static java.lang.annotation.ElementType.METHOD;"
+        + "import static java.lang.annotation.ElementType.PARAMETER;"
+        + "import static java.lang.annotation.RetentionPolicy.RUNTIME;"
+        + "import static java.lang.annotation.RetentionPolicy.SOURCE;"
+        + "@Retention(SOURCE)"
+        + "@Target({METHOD, PARAMETER, FIELD})"
+        + "public @interface Redacted {"
+        + "}");
+  }
+
+  @Test public void simple() {
+    JavaFileObject source = JavaFileObjects.forSourceString("test.Test", ""
+            + "package test;"
+            + "import com.google.auto.value.AutoValue;"
+            + "@AutoValue public abstract class Test {"
+            + "@Redacted public abstract String a();"
```

Comment:
One more thing! Include each of these three things but without `@Redacted` to verify normal behavior